src/store/wallet/effects/create/create.ts

```diff
@@ -728,3 +743,94 @@ export const getDecryptPassword =
       );
     });
   };
+
+export const detectAndCreateTokensForEachEvmWallet =
+  ({key, force}: {key: Key; force?: boolean}): Effect<Promise<void>> =>
+  async dispatch => {
+    try {
+      dispatch(
+        LogActions.info('starting [detectAndCreateTokensForEachEvmWallet]'),
+      );
+
+      const evmWalletsToCheck = key.wallets.filter(
```

Comment:
you can use: `!IsERCToken(currencyAbbreviation, chain) && IsEVMChain(chain)` for checking if it is a gas wallet

`const evmWalletsToCheck = key.wallets.filter((w)=>!IsERCToken(currencyAbbreviation, chain) && IsEVMChain(chain));`

Reply:
done